packages/web/src/services/FileDownloadServiceWeb.ts

```diff
@@ -1,23 +1,113 @@
+import axios from "axios";
+import StreamSaver from "streamsaver";
+
 import {
     FileDownloadService,
     DownloadResult,
     FileInfo,
     DownloadResolution,
-    HttpServiceBase,
+    FileStorageServiceBase,
 } from "../../../core/services";
 
-export default class FileDownloadServiceWeb extends HttpServiceBase implements FileDownloadService {
+export default class FileDownloadServiceWeb
+    extends FileStorageServiceBase
+    implements FileDownloadService {
     isFileSystemAccessible = false;
 
     public async download(fileInfo: FileInfo): Promise<DownloadResult> {
+        if (fileInfo.path.endsWith(".zarr")) {
+            const downloadResult = await this.handleZarrFile(fileInfo);
+            if (downloadResult) {
+                return downloadResult;
+            }
+        }
+
+        return this.downloadFile(fileInfo);
+    }
+
+    private async handleZarrFile(fileInfo: FileInfo): Promise<DownloadResult | null> {
```

Comment:
Maybe something like this? If we aren't using zarrita could this just download arbitrary folders?
```suggestion
    private async handleDirectoryLikeFile(fileInfo: FileInfo): Promise<DownloadResult | null> {
```